### PR TITLE
test: Do not sleep in sessionAffinity tests

### PR DESCRIPTION
**Pull request by brb**
Instead, wait until the deleted pod is removed from the cilium endpoint lists.

This is still non-ideal, as the removal does not mean that cilium-agent has processed k8s Endpoint event which removes the endpoint from the affinity match map. However, while running the changed test in the loop I haven't witnessed the flake.

### CODE REVIEW

**Comment by brb** [comment]
test-me-please

**Comment by nebril** [comment]
test-gke

File: test/helpers/kubectl.go [hunk 1]
```diff
@@ -569,6 +569,15 @@ func (kub *Kubectl) GetPodsNodes(namespace string, filter string) (map[string]st
 	return res.KVOutput(), nil
 }
 
+func (kub *Kubectl) GetPodNode(namespace, name string) (string, error) {
```
Comment (nebril) [inline test/helpers/kubectl.go]: nitpick: you can use `GetPodsNodes` instead of adding this function.

**Comment by brb** [comment]
test-with-kernel

**Comment by brb** [comment]
test-me-please

**Review by nebril — APPROVED** [review]

**Pull request merged**